### PR TITLE
#1506: apiVersion 2 with explicit data flow

### LINKED ISSUE #1506: Introduce brick `apiVersion: 2` with explicit data flow between bricks

### PR DESCRIPTION
**Pull request by twschiller**
Closes #1506

- [x] Update meta JSON Schema for bricks to support "v2"
- [x] Introduce an `explicitDataFlow` flag on runStage that's controlled by the apiVersion of the extension/composite brick
- [x] Default page editor to create bricks with v2 
- [x] Default to v1 behavior if the apiVersion is not specified

TODO:
- [ ] Update schema definitions in the API project

### CODE REVIEW

File: src/blocks/combinators.ts [hunk 1]
```diff
@@ -301,19 +334,41 @@ function arraySchema(itemSchema: Schema): Schema {
...
+  switch (version) {
+    case "v2": {
+      return { explicitDataFlow: true };
+    }
+
+    case "v1":
+    default: {
+      return { explicitDataFlow: false };
+    }
+  }
```
Comment (fregante) [inline src/blocks/combinators.ts:354]: Possibly:

```suggestion
  return { explicitDataFlow: version === "v2" };
```

File: src/devTools/editor/extensionPoints/base.ts [hunk 2]
```diff
@@ -57,6 +58,15 @@ export interface WizardStep {
   extraProps?: Record<string, unknown>;
 }
 
+/**
+ * Brick definition API controlling how the PixieBrix runtime interprets brick configurations
+ * @see ApiVersion
+ */
+export const PAGE_EDITOR_DEFAULT_BRICK_API_VERSION: ApiVersion = "v2";
```
Comment (fregante) [inline src/devTools/editor/extensionPoints/base.ts:65]: Is the default v2 or v1? I've seen v1 above

**Review by fregante — APPROVED** [review]
LGTM, however I'm not yet familiar with the data flow through the bricks. Hope to learn more in the next call

File: src/blocks/combinators.ts (hunk 1 above)
Comment (twschiller) [reply src/blocks/combinators.ts:354]: Yep -- I wrote it this way to keep version options sets explicit as we increment versions in the future and introduce more version-specific behavior

File: src/devTools/editor/extensionPoints/base.ts (hunk 2 above)
Comment (twschiller) [reply src/devTools/editor/extensionPoints/base.ts:65]: In the page editor it's defaulted to v2 for new bricks. It's defaulted to v1 if the apiVersion is not specified (to preserve backwards compatability)

**Pull request merged**